removing unused methods and validating required domain properties #1765

Amejia481 opened the pull request:
close #1765

Amejia481 committed:
removing unused methods and validating required domain properties #1765

Amejia481 committed:
Merge remote-tracking branch 'origin/bug-reviewing_operators' into bug-refactor_anemic_entities

# Conflicts:
#	app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java

Amejia481 committed:
Adding column reference to CountryVersionDB #1765

ifoche committed:
Merge branch 'development' into bug-refactor_anemic_entities

ifoche commented:
contains #1787 code, so waiting for that code to be finished before reviewing this one

xurxodev committed:
Merge branch 'development' into bug-refactor_anemic_entities

xurxodev reviewed (changes requested):
@Amejia481 I have found some changes to request you, but this issue finally will not be included in milestone E.1.1 release, so the next week we talk if you have doubts
Changes:
- Domain entities should not have public setter
- Domain entities should not receive a builder as dependency in constructor
You can look like autovalue in sdk to autogenerate  data class for having a example

Amejia481 committed:
Removing builders references from domain constructors
Removing public setters on domain entities

Amejia481 committed:
Refactoring method order dependency

Amejia481 committed:
Refactoring instrumentation test

xurxodev committed:
Merge branch 'development' into bug-refactor_anemic_entities

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Configuration.java:
```diff
@@ -16,75 +18,64 @@
-            setLastUpdate(builder.lastUpdate);
-            setReference(builder.reference);
-            setName(builder.name);
+            setUid(required(uid,"UID is required"));
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Sure!

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Country.java:
```diff
@@ -1,24 +1,17 @@
-    public Country() {
-    }
-
-    private Country(Builder builder) {
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Form.java:
```diff
@@ -1,22 +1,23 @@
+    public Form(long id, String name, int index, Type type,
+            Country country) {
 
-    private Form(Builder builder) {
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Header.java:
```diff
@@ -10,23 +12,14 @@
-    }
 
-    public Header() {
+        setId(required(id,"id is required"));
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Option.java:
```diff
@@ -132,23 +113,14 @@ public String toString() {
-        }
 
-        public Attribute() {
+            setId(id);
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java:
```diff
@@ -17,41 +19,24 @@
-        setIndex(builder.index);
-        setVisibility(builder.visibility);
-        setRules(builder.rules);
+        setId(required(id, "id is required"));
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java:
```diff
@@ -335,7 +330,7 @@ public static Builder newBuilder() {
         }
 
-        public void setConditions(
+        private void setConditions(
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java:
```diff
@@ -447,15 +438,15 @@ public String getValue() {
             }
 
-            public void setValue(String value) {
+            private void setValue(String value) {
```
Can you remove private setters? It's not necessary we have private setters, so we reduce the size of the class

Amejia481 replied:
Done!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Country.java:
```diff
@@ -29,46 +22,26 @@ public long getId() {
     }
 
-    @Override
-    public boolean equals(Object o) {
```
Why are you remove equals and hashcode methods? These methods can be useful in many comparison situations. If there is no reason, create them again, please.

Amejia481 replied:
I'm aware of the value that the equal method brings. I removed it because for the moment it hasn't been used it anywhere and one of the recommendations that I found on the internet for eliminating anemic entities was removing not used methods.

I will bring it back!

xurxodev commented on app/src/main/java/org/eyeseetea/malariacare/domain/entity/Form.java:
```diff
@@ -27,66 +28,42 @@ public long getId() {
     }
 
-    @Override
-    public boolean equals(Object o) {
```
Why are you remove equals and hashcode methods? These methods can be useful in many comparison situations. If there is no reason, create them again, please.

Amejia481 replied:
Done!

xurxodev commented on app/src/ereferrals/java/org/eyeseetea/malariacare/data/sync/importer/metadata/configuration/MetadataConfigurationDBImporter.java:
```diff
@@ -56,9 +56,8 @@ public MetadataConfigurationDBImporter(
-        if (!needToDownloadMetadata) {
-            throw new IllegalStateException("Has to call hasToUpdateMetadata before this method.");
-        }
+        fetchContriesVersionsIfRequired();
```
I don't understand this change in this issue, It's not related to refactoring anemic entities

Amejia481 replied:
Sorry for that, I took the advantage of this refactoring to break the order calling dependency between hasToUpdateMetadata and importMetadata methods.

Before this change, you were force to call hasToUpdateMetadata method before importMetadata method or you would got an exception, just like the text reads.
```
  throw new IllegalStateException("Has to call hasToUpdateMetadata before this method.");
```

xurxodev reviewed (changes requested):
@Amejia481 This is the begin of the way for this entities are not anemic but I have found some improvements to realize. Can you review it, please?

Amejia481 committed:
Removing private setter methods and unused setter methods.

-Adapting test QuestionConvertFromDomainVisitorShould to use
builder instead of setter method

Amejia481 committed:
Merge remote-tracking branch 'origin/bug-refactor_anemic_entities' into bug-refactor_anemic_entities

Amejia481 committed:
Small fix after merge

Amejia481 committed:
Adding equals, hashcode and equal to domain classes

Amejia481 commented:
I override equals, hashcode and toStrings methods to domain classes.

Didn't add equals and hashcode methods to the AppInfo, Configuration, Credentials, ForgotPasswordMessage, SurveyAnsweredRatio and UserAccount. I just added toString method because I'm not quite sure if they are going to be useful on those classes.

xurxodev committed:
Merge branch 'development' into bug-refactor_anemic_entities

xurxodev committed:
avoid dependencies between creation methods

xurxodev reviewed (approved):
thanks @Amejia481
